src/OpenTelemetry.Instrumentation.Wcf/Implementation/InstrumentedChannel.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="InstrumentedChannel.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.ServiceModel.Channels;
+
+namespace OpenTelemetry.Instrumentation.Wcf.Implementation;
+
+internal class InstrumentedChannel : InstrumentedCommunicationObject, IChannel
```

Comment:
nit: Would you please add `sealed` to these `internal` classes. There is a slight perf boost for sealed classes. See: https://github.com/dotnet/runtime/issues/49944

Reply:
Sorry for the long comment, but I think this is a useful nugget of information to spread around.  

I don't think this would apply here as the call site of the virtual methods would be WCF code which has no visibility into the concrete type. The situation where you get the performance improvement is this:
```c#
public abstract class FooBase
{
    public abstract void DoWork()
}

public class MyFoo : FooBase
{
    public override void DoWork()
    {
        Console.WriteLine("Doing some work");
    }
}

public static class Program
{
    public static int Main(string[] args)
    {
        MyFoo foo = GetFoo();
        foo.DoWork();
    }
    public static MyFoo GetFoo()
    {
        return new MyFoo();
    }
}
```
In the snippet above, the code in `Main` method which calls `GetFoo()` doesn't have knowledge about if the `MyFoo` instance returned is an actual `MyFoo` instance as it could potentially be a type derived from `MyFoo`. As the `DoWork()` method is virtual, it could potentially have been overridden in a derived instance so it must do a virtual method call to make sure the correct method is called. If `MyFoo` was sealed, then you are guaranteed that the object returned from `GetFoo()` is an actual `MyFoo` type as you can't derive it. This means the call to `DoWork()` can be a non-virtual method call to `MyFoo.DoWork()` as there's no possibility that a different method implementation would get called. The non-virtual method call is faster. In this case it would probably inline the implementation as `MyFoo.DoWork()` is so small which gives you even more performance improvement.  

For completeness and accuracy, I ignored that the compiler might be able to still do it in this case as `GetFoo()` is non-virtual and short so could get inlined in which case the compiler might reason about the type sufficiently to do a non-virtual method call here. The point was about semantics and if `GetFoo()` was longer so couldn't be inlined, or virtual (it's static in this code, but generalizing here), then what I stated would be correct. Hopefully you understand the concept I'm explaining in the general case.  

In this codebase, the instrumented channel implements multiple WCF interfaces. The caller for all the virtual overridden methods (or interface implemented methods) is WCF which only has a reference to the instance as the interface type. There is no a priori knowledge when WCF was compiled about the type of the channel so the compiler can only emit virtual method calls.  

You only benefit from sealing an implementation if the call site has a reference typed as the concrete sealed type.  

On the other hand, marking it as sealed doesn't do any harm. Although not applicable in this scenario, sealing a class can also prevent using Ref.Emit to derive an implementation and using reflection to inject it to modify an app to do things not intended.

Reply:
I've added `sealed` to all possible internal classes 👍